sdk/src/main/java/io/radar/sdk/util/RadarFileSystemLogBuffer.kt

```diff
@@ -0,0 +1,212 @@
+package io.radar.sdk.util
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import io.radar.sdk.Radar
+import io.radar.sdk.model.RadarLog
+import org.json.JSONObject
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.util.*
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * Wraps the filesystem to support log caching
+ */
+internal class RadarFileSystemLogBuffer(
+    /**
+     * Application Context
+     */
+    context: Context,
+    /**
+     * Executor for writing files. This is used to ensure that log entries are written to the file in the order that
+     * they are received.
+     */
+    private val executor: ExecutorService = Executors.newSingleThreadExecutor()
+) : RadarLogBuffer {
+
+    private companion object {
+
+        /**
+         * Name of the directory in which the logs are stored
+         */
+        const val DIRECTORY = "RadarLogs"
```

Comment:
what happens if there are multiple applications with the Radar SDK?

Reply:
This is the directory name within the app's internal file structure - not the whole Android phone.